File: 백준 문제 풀이 모음/python/class 4/9935.py
import sys
input = sys.stdin.readline

class stack :
    def __init__(self) :
        self.top = []
    def push(self, data) :
        self.top.append(data)
    def pop(self) :
        self.top.pop(-1)

stk = stack()

arr = input().rstrip()
target = input().rstrip()
l = len(target)

for i in arr :
    stk.push(i)
    if ''.join(stk.top[- l :]) == target :
        for _ in range(l) :
            stk.pop()

if stk.top :
    print(''.join(stk.top))
else :
    print('FRULA')

File: 백준 문제 풀이 모음/kotlin/class 4/9935.kt
import java.util.*

fun main() = with(System.`in`.bufferedReader()){
    val stk = Stack<Char>()

    val input1 = readLine().toCharArray()
    val input2 = readLine().toCharArray()
    val input2_final = input2[input2.size-1]

    for (i in 0 until input1.size) {
        val stk_input = input1[i]
        stk.add(stk_input)

        var count = 0
        if (input2_final.equals(stk_input)) {
            if (stk.size < input2.size) continue

            for (j in 0 until input2.size) {
                if (stk[stk.size - input2.size + j] == input2[j]) {
                    count++
                } else {
                    break
                }
            }
        }

        if (count == input2.size) {
            repeat(input2.size) {
                stk.pop()
            }
        }
//        println(stk)
    }

    val sb = StringBuilder()
    if (stk.empty()) {
        sb.append("FRULA")
    } else {
        stk.forEach {
            sb.append(it)
        }
    }
    print(sb)


}


